Tools/ardupilotwaf/ardupilotwaf.py

```diff
@@ -204,6 +204,26 @@ def ap_find_tests(bld, use=[]):
             cxxflags=['-Wno-undef'],
         )
 
+_versions = []
+
+@conf
+def ap_version_append(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, os.environ.get(k, v))]
+
+@conf
+def ap_version_append_str(ctx, k, v):
+    ctx.env['AP_VERSION_ITEMS'] += [(k, '"{}"'.format(os.environ.get(k, v)))]
```

Comment:
I left it there because we may want to add a define as a number and not as a string. I can remove it though.